### PR TITLE
Add part_channels method to Client

### PR DESCRIPTION
**Pull request by lemiffe**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature (compat with 1.2.2)

* **What is the current behavior?** (You can also link to an open issue here)
The bot can't part channels

* **What is the new behavior (if this is a feature change)?**
The bot can part channels

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

**Comment by EvieePy** [comment]
2.0/development doesn't currently have a `part_channels` on the connection.
I will keep this open though as I can implement part_channels on the connection object which will allow this to work.

File: twitchio/client.py [hunk 1]
```diff
@@ -262,6 +262,16 @@ async def join_channels(self, channels: Union[List[str], Tuple[str]]):
         """
         await self._connection.join_channels(*channels)
 
+    async def part_channels(self, channels: Union[List[str], Tuple[str]]):
+        """|coro|
+        Part the specified channels.
```
Comment (EvieePy) [inline twitchio/client.py:267]: Should be a newline here,

File: twitchio/client.py [hunk 2]
```diff
@@ -262,6 +262,16 @@ async def join_channels(self, channels: Union[List[str], Tuple[str]]):
...
+        Parameters
+        ------------
+        channels: Union[List[str], Tuple[str]]
+            The channels in either a list or tuple form to join.
```
Comment (EvieePy) [inline twitchio/client.py:271]: Should be part not join.

File: twitchio/client.py (hunk 1 above)
Comment (lemiffe) [reply twitchio/client.py:267]: ✅ Done

**Comment by lemiffe** [comment]
PR remarks done :) just missing implementation in the connection I think @EvieePy

**Comment by EvieePy** [comment]
Sorry didn't realize this was on the development branch and was auto closed and never really looked at since we merged into master.

**Pull request closed without merging**